src/modules/mc_pos_control/MulticopterPositionControl.cpp

```diff
@@ -223,17 +149,41 @@ void MulticopterPositionControl::Run()
 
 	perf_begin(_cycle_perf);
 
-	if (_local_pos_sub.update(&_local_pos)) {
+	parameters_update(false);
+
+	// require at least flag_control_climb_rate_enabled to run
+	_control_mode_sub.update(&_control_mode);
+
+	vehicle_local_position_s local_pos;
 
-		poll_subscriptions();
-		parameters_update(false);
+	if (_control_mode.flag_control_climb_rate_enabled && _local_pos_sub.update(&local_pos)) {
+
+		const hrt_abstime &time_stamp_now = local_pos.timestamp_sample;
+
+		const Vector3f velocity{local_pos.vx, local_pos.vy, local_pos.vz};
+
+		if (time_stamp_now - _time_stamp_last_loop > 500_ms) {
+			_time_stamp_last_loop = local_pos.timestamp_sample - SCHEDULE_INTERVAL_US;
+			_velocity = velocity;
+			_vel_deriv_lpf.reset(Vector3f{local_pos.ax, local_pos.ay, local_pos.az});
```

Comment:
As far as I understand this replaces the BlockDerivative reset when the position controller did not run but it also needs to be reset when the velocity estimate just got valid e.g. was NAN before. See https://github.com/PX4/PX4-Autopilot/pull/13922